golink.go

```diff
@@ -366,9 +366,9 @@ func serveGo(w http.ResponseWriter, r *http.Request) {
 	stats.dirty[link.Short]++
 	stats.mu.Unlock()
 
-	currentUser, _ := currentUser(r)
+	login, _ := currentUser(r)
 
-	target, err := expandLink(link.Long, expandEnv{Now: time.Now().UTC(), Path: remainder, User: currentUser})
+	target, err := expandLink(link.Long, expandEnv{Now: time.Now().UTC(), Path: remainder, User: login})
```

Comment:
I worry about undefined behavior in the no-login case if the golink expansion expects a user.

Should we return a well-known error during expansion if the `expandEnv.User` field is zero value and the template calls for it?

Reply:
done.  I broke this PR into two parts to make it easier to review.  Adding tests and a little refactoring (with no behavioral changes) is now in #73.  This PR just adds the "unknown user" functionality, but additionally puts it behind a flag now.